app/scripts/controllers/swaps.js

```diff
@@ -71,6 +71,7 @@ const initialState = {
     topAggId: null,
     routeState: '',
     swapsFeatureIsLive: false,
+    isFeatureFlagLoaded: false,
```

Comment:
Does this need to be background state? :thinking:  It seems to only be used in one UI component. Maybe it could be migrated to component state - set to false when the component mounts, and true when the `fetchSwapsLiveness` call resolves. That would make it easier to use the loading screen in more cases as well.

Reply:
That's right that it's only used in one UI component. What do you think about my code below? I'm using a recommended way to make async calls from the `useEffect` function: https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret

```
const [isFeatureFlagLoaded, setIsFeatureFlagLoaded] = useState(false);

useEffect(() => {
    const fetchSwapsLivenessWrapper = async () => {
      await dispatch(fetchSwapsLiveness());
      setIsFeatureFlagLoaded(true);
    };
    fetchSwapsLivenessWrapper();
  }, []);
```

Reply:
`useEffect` certainly seems appropriate!

The pattern you're using here with a "wrapper" function seems a bit odd. I don't follow what the wrapper is for :thinking:

Reply:
Without it React shows a warning:
![image](https://user-images.githubusercontent.com/80175477/114227021-45483100-9929-11eb-9a74-0e7b4330de20.png)

When I use the wrapper, no warning :)

Reply:
Got it, that makes sense.